benchmarks/random_allocations/random_allocations.cpp

```diff
@@ -315,35 +254,41 @@ int main(int argc, char** argv)
     if (argc > 3) max_size = atoi(argv[3]);
     declare_benchmark(mr_name);
   } else {
-    BENCHMARK_TEMPLATE(BM_RandomAllocations, pool_mr)->Apply(benchmark_range);
-    BENCHMARK_TEMPLATE(BM_RandomAllocations, safe_hybrid_mr)->Apply(benchmark_range);
-    BENCHMARK_TEMPLATE(BM_RandomAllocations, cnmem_mr)->Apply(benchmark_range);
-    BENCHMARK_TEMPLATE(BM_RandomAllocations, cuda_mr)->Apply(benchmark_range);
+    std::vector<std::string> mrs{"pool", "hybrid", "cnmem", "cuda"};
+    std::for_each(mrs.cbegin(), mrs.cend(), [](auto const& s) { declare_benchmark(s); });
   }
 
   ::benchmark::RunSpecifiedBenchmarks();
 }
 
-// For profiling
-/*template <typename MemoryResource>
-static void RandomAllocations(size_t num_allocations, size_t max_size) {
-  MemoryResource mr;
+/*
+// For profiling, don't remove
+static void RandomAllocations(MRFactoryFunc factory, size_t num_allocations, size_t max_size)
+{
+  auto mr = factory();
 
   try {
-    uniform_random_allocations(mr, num_allocations, max_size, max_usage);
+    uniform_random_allocations(*mr, num_allocations, max_size, max_usage);
   } catch (std::exception const& e) {
     std::cout << "Error: " << e.what() << "\n";
   }
 }
 
-int main(int argc, char** argv) {
-  std::string name{ argc > 1 ? argv[1] : "cnmem" };
+int main(int argc, char** argv)
+{
+  std::string name{argc > 1 ? argv[1] : "cnmem"};
+
+  constexpr size_t n        = 100000;
+  constexpr size_t max_size = 4096;
 
   if (name == "hybrid")
-    RandomAllocations<rmm::mr::hybrid_memory_resource>(1000, 4096);
-  else if (name == "sub")
-    RandomAllocations<rmm::mr::pool_memory_resource>(1000, 4096);
+    RandomAllocations(&make_hybrid, n, max_size);
+  else if (name == "pool")
+    RandomAllocations(&make_pool, n, max_size);
   else if (name == "cnmem")
-    RandomAllocations<rmm::mr::cnmem_memory_resource>(1000, 4096);
+    RandomAllocations(&make_cnmem, n, max_size);
+
+  std::cout << "Finished\n";
   return 0;
-}*/
+}
+*/
```

Comment:
Thanks for the reminder. Actually I plan to make a command line argument to run the profiling path instead of the benchmarking path (profiling just runs the benchmark once rather than multiple times for a simpler profile).

Reply:
Done